onnxscript/rewriter/pattern_test.py

```diff
@@ -780,6 +780,189 @@ def test_model1(x: FLOAT[16, 32], y: FLOAT[16, 32]) -> FLOAT[16, 32]:
         self.assertEqual([x.op_type for x in model.graph], ["ReluPlus"])
 
 
+class ValueNodeCheckersTest(unittest.TestCase):
+    """Test value/node level checkers functionality."""
+
+    def test_value_pattern_with_check(self):
+        """Test ValuePattern with check attribute."""
+        from onnxscript.rewriter import _pattern_ir
+        
+        def value_checker(context, value):
+            return True
+        
+        # Test creating ValuePattern with check
+        value_pattern = _pattern_ir.ValuePattern("test_value", check=value_checker)
+        self.assertEqual(value_pattern._check, value_checker)
+        self.assertEqual(value_pattern.name, "test_value")
+
+    def test_node_pattern_with_check(self):
+        """Test NodePattern with check attribute."""
+        from onnxscript.rewriter import _pattern_ir
+        
+        def node_checker(context, node):
+            return True
+        
+        # Test creating NodePattern with check
+        domain_pattern = _pattern_ir.StringConstantPattern("")
+        inputs = []
+        attributes = {}
+        outputs = ["output"]
+        
+        node_pattern = _pattern_ir.NodePattern(
+            domain_pattern,
+            "Add",
+            inputs,
+            attributes,
+            outputs,
+            allow_other_attributes=True,
+            allow_other_inputs=True,
+            check=node_checker
+        )
+        self.assertEqual(node_pattern._check, node_checker)
+
+    def test_to_value_pattern_with_callable(self):
+        """Test _to_value_pattern function with callable input."""
+        from onnxscript.rewriter import _pattern_ir
+        
+        def my_checker(context, value):
+            return True
+        
+        result = _pattern_ir._to_value_pattern(my_checker)
+        self.assertIsInstance(result, _pattern_ir.ValuePattern)
+        self.assertEqual(result._check, my_checker)
+        self.assertIsNone(result.name)
+
+    def test_op_pattern_builder_with_check(self):
+        """Test OpPatternBuilder with _check parameter."""
+        from onnxscript.rewriter import _pattern_ir
+        
+        def node_checker(context, node):
+            return True
+        
+        # Create OpPatternBuilder
+        opset_builder = _pattern_ir.OpsetPatternBuilder("")
+        op_builder = opset_builder.Add
+        
+        # Call with _check parameter
+        result = op_builder(None, None, _check=node_checker)
+        
+        # The result should be a NodeOutputPattern, and its producer should have the check
+        self.assertTrue(hasattr(result, 'producer'))
+        producer = result.producer()
+        self.assertIsNotNone(producer)
+        self.assertTrue(hasattr(producer, '_check'))
+        self.assertEqual(producer._check, node_checker)
+
```

Comment:
Can you move the newly introduced test cases above, which are about _pattern_ir into a new _pattern_ir_test.py file?